### PR TITLE
feat: add Swift codes for chapter_sorting articles

### PR DESCRIPTION
**Pull request by nuomi1**
If this PR is related to coding or code translation, please fill out the checklist.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

**Comment by nuomi1** [comment]
```text
result of `bubble_sort.swift`

冒泡排序完成后 nums = [1, 1, 2, 3, 4, 5]
冒泡排序完成后 nums1 = [1, 1, 2, 3, 4, 5]
```

```text
result of `insertion_sort.swift`

插入排序完成后 nums = [1, 1, 2, 3, 4, 5]
```

```text
result of `quick_sort.swift`

快速排序完成后 nums = [0, 1, 2, 3, 4, 5]
快速排序（中位基准数优化）完成后 nums1 = [0, 1, 2, 3, 4, 5]
快速排序（尾递归优化）完成后 nums2 = [0, 1, 2, 3, 4, 5]
```

```text
result of `merge_sort.swift`

归并排序完成后 nums = [0, 1, 2, 3, 4, 5, 6, 7]
```

```text
result of `radix_sort.swift`

基数排序完成后 nums = [3, 4, 12, 23, 192, 788]
```

**Review by krahets — COMMENTED** [review]
Awesome! Please address the comments.

File: codes/swift/chapter_sorting/quick_sort.swift [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    quickSort(nums: &nums, left: left, right: pivot - 1)
+    quickSort(nums: &nums, left: pivot + 1, right: right)
+}
+
+/* 快速排序类（中位基准数优化） */
+// 快速排序类（中位基准数优化）-选取三个元素的中位数
+func quickSortMedianThree(nums: [Int], left: Int, mid: Int, right: Int) -> Int {
+    // 使用了异或操作来简化代码
+    // 异或规则为 0 ^ 0 = 1 ^ 1 = 0, 0 ^ 1 = 1 ^ 0 = 1
+    if (nums[left] < nums[mid]) ^ (nums[left] < nums[right]) {
```
[the remark this answers is not in the document]
Comment (nuomi1) [reply codes/swift/chapter_sorting/quick_sort.swift]: Done.

**Review by krahets — APPROVED** [review]
Thanks!

**Pull request merged**